navigation/navigation-compose-material/integration-tests/navigation-demos/src/main/java/androidx/navigation/compose/material/demos/BottomSheetNavDemo.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.navigation.compose.material.demos
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.Button
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.material.ModalBottomSheetLayout
+import androidx.navigation.compose.material.bottomSheet
+import androidx.navigation.compose.material.rememberBottomSheetNavigator
+import androidx.navigation.compose.navigate
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.plusAssign
+import java.util.UUID
+
+private object Destinations {
+    const val Home = "HOME"
+    const val Feed = "FEED"
+    const val Sheet = "SHEET"
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun BottomSheetNavDemo() {
+    val navController = rememberNavController()
+    val bottomSheetNavigator = rememberBottomSheetNavigator()
+    navController.navigatorProvider += bottomSheetNavigator
```

Comment:
This is going to cause issues if this method recomposes. Perhaps what we actually want here is a `rememberNavController()` overload (/optional parameter) that takes a `NavHostController.() -> Unit` lambda where we could do this type of configuration when the NavController is created?

I'm not sure how we'd effectively match the `remember` scoping of the `NavController` and the `BottomSheetNavigator`, as generally we'd only want those to be 1:1. Perhaps not really an issue since the `remember` is ~unconditional.

Reply:
Yeah, good catch! I think an optional parameter would be nice.

```kotlin
val bottomSheetNavigator = rememberBottomSheetNavigator()
val navController = rememberNavController {
    navigatorProvider += bottomSheetNavigator
}
```

```kotlin
@Composable
fun rememberNavController(
    configureNavController: NavHostController.() -> Unit = { }
): NavHostController {
    val context = LocalContext.current
    return rememberSaveable(configureNavController, saver = NavControllerSaver(context)) {
        createNavController(context).apply(configureNavController)
    }
}
```

I don't think that scoping is a big issue here, and making `configureNavController` `@Composable` isn't great either imo. In case it's an issue, we'll see it pop up in alpha I think :)